x-pack/plugins/ml/public/alerting/preview_alert_condition.tsx

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useCallback, useEffect, useMemo, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCode,
+  EuiDescriptionList,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiHorizontalRule,
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import type { AlertingApiService } from '../application/services/ml_api_service/alerting';
+import { MlAnomalyDetectionAlertParams, PreviewResponse } from '../../common/types/alerts';
+import { composeValidators } from '../../common';
+import { requiredValidator, timeIntervalInputValidator } from '../../common/util/validators';
+import { invalidTimeIntervalMessage } from '../application/jobs/new_job/common/job_validator/util';
+
+export interface PreviewAlertConditionProps {
+  alertingApiService: AlertingApiService;
+  alertParams: MlAnomalyDetectionAlertParams;
+}
+
+const AlertInstancePreview: FC<PreviewResponse['results'][number]> = React.memo(
```

Comment:
is it worth moving this component into a separate file?

Reply:
For me it seems like a fairly small component only relevant to the alert preview. But no problem with extracting it to a separate file.